conda_smithy/configure_feedstock.py

```diff
@@ -305,7 +307,41 @@ def _yaml_represent_ordereddict(yaml_representer, data):
     return yaml.representer.SafeRepresenter.represent_dict(yaml_representer, data.items())
 
 
-def dump_subspace_config_files(metas, root_path, output_name):
+def finalize_config(config, platform):
+    """Specialized handling to deal with the dual compiler output state.
+    In a future state this SHOULD go away"""
+    # TODO: REMOVE WHEN NO LONGER NEEDED
+    if platform in {'linux', 'osx'}:
+        if len({'c_compiler', 'cxx_compiler', 'fortran_compiler'} & set(config.keys())):
+            # we have a compiled source here so the zip should take care of things appropriately
+            pass
+        else:
+            try:
+                # prefer to build with the newer compiler image, This ensures that for things that don't declare they need
+                # compilers, they will fail
+                config['docker_image'] = [config['docker_image'][-1]]
+            except KeyError:
+                config['docker_image'] = ['condaforge/linux-anvil']
+
+            try:
+                config['channel_sources'] = [config['channel_sources'][0]]
+            except KeyError:
+                config['channel_sources'] = ['conda-forge,defaults']
```

Comment:
It looks like this is a comma separated list within a YAML list. Should this also be a YAML list?